### PR TITLE
ROC-6393: Link defendant to claim test support endpoint

### PR DESCRIPTION
**Pull request by daveoh**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/LinkDefendantToClaimViaAuthenticationTest.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                .withUserId(DEFENDANT_ID)
+                .withMail(DEFENDANT_EMAIL)
+                .withRoles("citizen", "letter-" + SampleClaim.LETTER_HOLDER_ID)
+                .build())
+            .build()
+        );
+
+        Claim claim = claimStore.saveClaim(SampleClaimData.builder().build());
+
+        String urlTemplate = String.format("/testing-support/claims/%s/defendant/%s/%s",
```
Comment (tharacka) [inline src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/LinkDefendantToClaimViaAuthenticationTest.java]: "/testing-support/claims/%s/defendant/%s/%s" can this be static variable as it is repeated through out the tests?

Comment (tharacka) [inline src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/LinkDefendantToClaimViaAuthenticationTest.java]: declare /testing-support/claims/%s/defendant/%s/%s as static variable due to repetition?

File: src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/IntegrationTestSupportController.java [hunk 2]
```diff
@@ -75,6 +85,27 @@ public void linkDefendantToClaim(
...
+    @PutMapping("/claims/{claimReferenceNumber}/defendant/{defendantUsername}/{defendantPassword}")
+    @ApiOperation("Link a claim to a defendant")
+    public void linkDefendantToClaim(
+        @PathVariable("claimReferenceNumber") String claimReferenceNumber,
+        @PathVariable("defendantUsername") String defendantUsername,
+        @PathVariable("defendantPassword") String defendantPassword
+    ) {
+        logger.info("Linking claim to defendant");
+            Claim claim = getClaim(claimReferenceNumber, null);
+        try {
```
Comment (dharmendrak) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/IntegrationTestSupportController.java]: please take away try catch, are we are just catching to log the message, technically you see that in app insight.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/IntegrationTestSupportController.java [hunk 3]
```diff
@@ -23,16 +27,22 @@
 @ConditionalOnProperty("claim-store.test-support.enabled")
 public class IntegrationTestSupportController {
 
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
```
Comment (dharmendrak) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/IntegrationTestSupportController.java]: we don't need logging here, unless if you say so has valid reason.

**Pull request merged**